src/rust/engine/fs/store/src/lib.rs

```diff
@@ -267,11 +267,84 @@ impl RemoteStore {
           }
           local_store
             .store_bytes(entry_type, digest.hash, bytes, true)
-            .await?;
+                .await?;
+            }
+            Ok(())
+        })
+        .await
+    }
+
+    async fn download_digest_to_local_batch(
```

Comment:
The singular version of this, `download_digest_to_local` employs some deduplication via 

https://github.com/pantsbuild/pants/blob/2f116606f356866f2efb425295106a80a9282149/src/rust/engine/fs/store/src/lib.rs#L205-L207 

How do we want to handle this for the batch version?

Reply:
Thinking out loud, there's potentially two levels of deduplication:

- globally (if there's two concurrent `download_digest_to_local_batch` or a `download_digest_to_local_batch` and `download_digest_to_local` call with the same digest)
- locally (a single `download_digest_to_local_batch` where the same digest appears more than once)

The latter could be solved by deduplicating the `digests` input here, but seems of relatively low value.

Keeping the "start simple" approach, I'd be inclined to do nothing in this PR, and assume that it is fine and can/should be solved with the more complicated "run a background actor that collects requests and batches them internally" (i.e. deduplicate at that level) if we need that.